src/core.js

```diff
@@ -1,6 +1,44 @@
 'use strict';
+import {Model} from './model';
+import {serialize, deserialize} from './serializers/xml';
+import {persist} from './adapters/mediawiki';
+
+export {Core as InfoboxTemplateBuilder};
```

Comment:
Yep, they are hoisted in ES6, and imports/exports at the top seems to be the de facto best practice. Note, class definitions are not hoisted in TypeScript.